src/main/scala/no/gdl/bookapi/model/domain/Chapter.scala

```diff
@@ -8,14 +8,34 @@
 package no.gdl.bookapi.model.domain
 
 import no.gdl.bookapi.BookApiProperties
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Entities.EscapeMode
+import org.jsoup.select.Elements
 import scalikejdbc._
 
 case class Chapter(id: Option[Long],
                    revision: Option[Int],
                    translationId: Long,
                    seqNo: Int,
                    title: Option[String],
-                   content: String)
+                   content: String) {
+
+  def imagesInChapter(): Seq[Long] = {
+    import com.netaporter.uri.dsl._
+
+    val document = Jsoup.parseBodyFragment(content)
+    val images: Elements = document.select("embed[data-resource='image']")
+    var imageIds: Seq[Long] = Seq()
+
+    for (i <- 0 until images.size()) {
+      val image = images.get(i)
+      val nodeId = image.attr("data-resource_id")
+      imageIds = imageIds :+ nodeId.toLong
+    }
+
+    imageIds
+  }
```

Comment:
Instead of mutating `imageIds`, can this be done with a for comprehension? Something like
```scala
    val imageIds = for {
      i <- 0 until images.size()
      image = images.get(i)
      nodeId = image.attr("data-resource_id")
    } yield nodeId.toLong
    imageIds
```
?

Reply:
Nice!
This is a far better solution.